packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2522,6 +2526,37 @@ function warnIfNotCurrentlyActingUpdatesInDEV(fiber: Fiber): void {
 
 export const warnIfNotCurrentlyActingUpdatesInDev = warnIfNotCurrentlyActingUpdatesInDEV;
 
+// in tests, we want to enforce a mocked scheduler.
+let didWarnAboutUnmockedScheduler = false;
+
+export function warnIfUnmockedScheduler(fiber: Fiber) {
+  if (__DEV__) {
+    if (didWarnAboutUnmockedScheduler === false) {
+      if (fiber.mode & BatchedMode || fiber.mode & ConcurrentMode) {
```

Comment:
should we trigger the warning for legacy sync mode + strict mode too?

Reply:
I think it's a good idea to do so, since folks will use strict mode as a stepping stone to concurrent mode. changing.

Reply:
maybe not. and we can't tell when rendering if a child tree will be in strict mode.